build(common): specify explicit locales dir in package.json

filipesilva opened the pull request:
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)


## PR Type
What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->
```
[x] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] angular.io application / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?
Locale file directory is implicit in `@angular/common`.

## What is the new behavior?
Locale file directory is explicit in `@angular/common`.

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications below. -->
Followup from #19474.

filipesilva committed:
build(common): specify explicit locales dir in package.json

mary-poppins commented:
You can preview f5497e5 at https://pr21016-f5497e5.ngbuilds.io/.

vicb commented on packages/common/test/i18n/locale_data_folder.ts:
```diff
@@ -0,0 +1,23 @@
+import {basename, resolve} from 'path';
+
+export function main() {
+  describe('locale data folder', () => {
```
Don't we filter test files with `*_spec.ts` ? (ie is this spec executed at all ?)

filipesilva replied:
Right you are, my bad. Fixing.

vicb reviewed (changes requested):
Please rename the file to be consistent and make sure the test is executed.

vicb commented:
Sound like a `build` commit to me more than a `fix` - as this is not fixing anything. 
As such is there a real interest in merging in in the patch branch (given tools should support earlier version anyway) ?

filipesilva commented:
@vicb I don't think it is important to merge to patch, no. In a minor would be fine.

I've changed the commit message and renamed the spec file, but I'm not sure what's going on with the pullapprove check.

filipesilva commented:
@vicb I'm doing a unit test right now but that's the wrong approach. This sort of stuff is more of an integration test. The only package that seems to have integration tests right now is `compiler-cli` though.

Should I make a new script for `common` package integration tests? Or is there another approach you would rather me follow.

filipesilva commented:
@ocombe also mentioned `./integration/i18n/` as a good place to add a test (via a node script).

vicb reviewed (approved):
Thanks for the updates

mary-poppins commented:
You can preview f5497e5 at https://pr21016-f5497e5.ngbuilds.io/.
You can preview 0725767 at https://pr21016-0725767.ngbuilds.io/.
You can preview 589373f at https://pr21016-589373f.ngbuilds.io/.
You can preview 0553690 at https://pr21016-0553690.ngbuilds.io/.

mary-poppins commented:
You can preview deced02 at https://pr21016-deced02.ngbuilds.io/.

filipesilva commented:
I followed @ocombe's suggestion to add an integration test instead, @vicb PTAL.

ocombe commented:
Sounds good to me

vicb commented:
@filipesilva remember to add the "merge" label for PRs to be merged !

mhevery commented:
http://test/OCL:183314180:BASE:183314216:1516928283330:35ca6a6f

filipesilva commented:
@vicb I cannot add labels in angular/angular :/
